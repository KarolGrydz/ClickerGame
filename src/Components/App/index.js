import React, { Component } from 'react';
import { Container, Col } from 'reactstrap';
import styled from 'styled-components';
import Footer from '../Footer';
import Header from '../Header';
import Home from '../Home';
import Firebase from '../Firebase';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

class App extends Component {
  state = {
    players: [],
    addScore: false,
    numberOfClickToAdd: 0,
    playerExist: true,
    firebase: new Firebase()
  };

  addNewScore = numberOfClickToAdd => this.setState({ numberOfClickToAdd });

  changeSite = () => {
    let { addScore } = this.state;
    addScore = !addScore;
    this.setState({ addScore });
  };

  gotData = data => {
    let newPlayers = [];
    let scores = data.val();
    let keys = Object.keys(scores);
    for (let i = 0; i < keys.length; i++) {
      let k = keys[i];
      let name = scores[k].name;
      let points = scores[k].points;
      let newPlayer = {
        id: k,
        name,
        points
      };

      newPlayers.push(newPlayer);
    }

    newPlayers.sort((a, b) => b.points - a.points);
    newPlayers = newPlayers.splice(0, 10);

    this.setState({
      players: newPlayers
    });
  };

  gotError = err => console.log('Error ', err);

  componentDidMount() {
    const { firebase } = this.state;
    firebase.getDataFromFireBase(this.gotData, this.gotError);
  }

  render() {
    const { firebase } = this.state;
    let { players, addScore, numberOfClickToAdd } = this.state;
    return (
      <Container fluid={true} className="text-center">
        <Col md="12" sm="12" xs="12">
          <MainPage>
            <Header addScore={addScore} />
            <Home
              addScore={addScore}
              players={players}
              addNewScore={this.addNewScore}
              changeSite={this.changeSite}
              numberOfClickToAdd={numberOfClickToAdd}
              addScoreToFireBase={firebase.addScoreToFireBase}
              changeAddScore={this.changeAddScore}
            />
          </MainPage>
          <Footer players={players} />
        </Col>
      </Container>
    );
  }
}

export default App;

const MainPage = styled.div`
  background-color: #eee;
  padding-top: 48px;
  padding-bottom: 48px;
  text-align: center;
`;

export const Title = styled.h1`
  font-size: 4.5rem;
  font-weight: 300;
`;

export const Paragraf = styled.p`
  font-size: 1.4rem;
  font-weight: 200;
  margin-bottom: 15px;
`;
