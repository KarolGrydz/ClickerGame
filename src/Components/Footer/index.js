import React from 'react';
import { Col } from 'reactstrap';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Modals from '../Modals';

const Footer = players => (
  <FooterDiv>
    <Col md="12" sm="12" xs="12" className="text-center">
      <Modals players={players} />
    </Col>
    <a href="https://kg.glogow.pl">English version</a>
    <p>
      Stworzone przez <a href="https://kg.glogow.pl">Ranald</a> 2016, Wszelkie
      prawa zastrze≈ºone
    </p>
  </FooterDiv>
);

Footer.propTypes = {
  players: PropTypes.array.isRequired
};

export default Footer;

const FooterDiv = styled.div`
  padding-top: 50px;
  font-size: 0.75rem;
  background-color: #fff;
`;
