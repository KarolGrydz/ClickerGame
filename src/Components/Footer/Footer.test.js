import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Title } from '../App';
import Footer from './index';

describe('<Footer />', () => {
  it('should be defined', () => {
    expect(Footer).toBeDefined();
  });
  it('should render without crashing', () => {
    const footer = shallow(<Footer />);
    expect(shallowToJson(footer)).toMatchSnapshot();
  });
  it('props players should be array', () => {
    const header = shallow(<Footer addScore={false} />);
    const addScore = true;
    header.setProps({ addScore });
    expect(header.contains(<Title>Dodaj sw√≥j wynik</Title>)).toBeTruthy();
    expect(shallowToJson(header)).toMatchSnapshot();
  });
});
