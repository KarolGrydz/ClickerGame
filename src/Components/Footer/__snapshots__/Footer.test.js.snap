// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render without crashing 1`] = `
<styled.div>
  <Col
    className="text-center"
    md="12"
    sm="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <Modals
      players={Object {}}
    />
  </Col>
  <a
    href="https://kg.glogow.pl"
  >
    English version
  </a>
  <p>
    Stworzone przez 
    <a
      href="https://kg.glogow.pl"
    >
      Ranald
    </a>
     2016, Wszelkie prawa zastrze≈ºone
  </p>
</styled.div>
`;
